### ISSUE TITLE
--with-gnutls --with-openssl enables USE_TLS_SRP, even when OpenSSL/LibreSSL doesn't support it; build fails

### ISSUE THREAD

**Issue by steils** [curl/curl#5865]
`./configure --with-ssl --with-nettle --with-gnutls --with-openssl`

[configure.log](https://github.com/curl/curl/files/5132432/configure.log)

`checking for SRP_Calc_client_key in -lcrypto... no`
`checking for gnutls_srp_verifier in -lgnutls... yes`
`checking whether to enable TLS-SRP authentication... yes`

Then `make`...
```
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H   -I../include -I../lib -I../lib  -DBUILDING_LIBCURL  -DCURL_HIDDEN_SYMBOLS   -fvisibility=hidden -Werror-implicit-function-declaration -O2 -Wno-system-headers -pthread  -MT vtls/libcurl_la-openssl.lo -MD -MP -MF vtls/.deps/libcurl_la-openssl.Tpo -c -o vtls/libcurl_la-openssl.lo `test -f 'vtls/openssl.c' || echo './'`vtls/openssl.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I../include -I../lib -I../lib -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -fvisibility=hidden -Werror-implicit-function-declaration -O2 -Wno-system-headers -pthread -MT vtls/libcurl_la-openssl.lo -MD -MP -MF vtls/.deps/libcurl_la-openssl.Tpo -c vtls/openssl.c  -fPIC -DPIC -o vtls/.libs/libcurl_la-openssl.o
vtls/openssl.c: In function ‘ossl_connect_step1’:
vtls/openssl.c:2802:9: error: implicit declaration of function ‘SSL_CTX_set_srp_username’; did you mean ‘SSL_CTX_set_ssl_version’? [-Werror=implicit-function-declaration]
 2802 |     if(!SSL_CTX_set_srp_username(backend->ctx, ssl_username)) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
      |         SSL_CTX_set_ssl_version
vtls/openssl.c:2806:9: error: implicit declaration of function ‘SSL_CTX_set_srp_password’; did you mean ‘SSL_CTX_set_ssl_version’? [-Werror=implicit-function-declaration]
 2806 |     if(!SSL_CTX_set_srp_password(backend->ctx, SSL_SET_OPTION(password))) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
      |         SSL_CTX_set_ssl_version
cc1: some warnings being treated as errors
make[2]: *** [Makefile:2611: vtls/libcurl_la-openssl.lo] Error 1
```

It's LibreSSL 3.2.1, but the version is irrelevant, because SRP was removed from LibreSSL in 2014.

**Comment by emilengler** [comment]
I'm curious if two TLS back ends are even allowed, it would make no sense

**Comment by steils** [comment]
@emilengler it should make sense, see https://github.com/curl/curl/pull/2665

**Comment by bagder** [comment]
@emilengler to allow the application to select the specific TLS backend in run-time rather than just one at build time. Supported since libcurl 7.56.0. See `curl_global_sslset()`.

**Comment by bagder** [comment]
@steils please try #5870 and see if that fixes this issue for you!

**Comment by GitGangGuy** [comment]
Hello @bagder, I really appreciate you taking your time to fix this, but I'm trying to compile curl on Gentoo (from the official repository) with **libressl** as the ssl provider and I'm experiencing the same issue (extract from the build log after compilation failed):
```
 * SSL provided by gnutls
 * SSL provided by openssl
 * Default SSL provided by LibreSSL
[...]
checking whether to enable TLS-SRP authentication... yes
[...]
/var/tmp/portage/net-misc/curl-7.72.0/work/curl-7.72.0/lib/vtls/openssl.c:2802:9: error: implicit declaration of function ‘SSL_CTX_set_srp_username’; did you mean ‘SSL_CTX_set_ssl_version’? [-Werror=implicit-function-declaration]
 2802 |     if(!SSL_CTX_set_srp_username(backend->ctx, ssl_username)) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
      |         SSL_CTX_set_ssl_version
/var/tmp/portage/net-misc/curl-7.72.0/work/curl-7.72.0/lib/vtls/openssl.c:2806:9: error: implicit declaration of function ‘SSL_CTX_set_srp_password’; did you mean ‘SSL_CTX_set_ssl_version’? [-Werror=implicit-function-declaration]
 2806 |     if(!SSL_CTX_set_srp_password(backend->ctx, SSL_SET_OPTION(password))) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
      |         SSL_CTX_set_ssl_version
cc1: some warnings being treated as errors
make[2]: *** [Makefile:2615: vtls/libcurl_la-openssl.lo] Error 1
```
I saw that the PR you opened hasn't been merged. Was this intentional? Are these changes going to make it into production anytime?

**Comment by bagder** [comment]
See above. This was fixed in commit 68a5132 which is not part of 7.72.0 but is part of 7.73.0...

**Comment by jzakrzewski** [comment]
> I saw that the PR you opened hasn't been merged.

You saw wrong.

> bagder closed this in 68a5132 on Aug 28
